src/components/AddComment.jsx

```diff
@@ -0,0 +1,65 @@
+import { useState } from "react";
+import { postComment } from "../utils/api";
+import { useParams } from "react-router-dom";
+import CommentList from "./CommentList";
+
+const AddComment = ({ setComments }) => {
+  const [author, setAuthor] = useState("");
+  const [body, setBody] = useState("");
+  const [check, setCheck] = useState(false);
+  const { article_id } = useParams();
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if(body!=="" && author!=="") {
+        const optimisticComment = {
+          author: author,
+          body: body,
+          votes: 0,
+        };
+        const newComment = {
+          author: author,
+          body: body,
+        };
+        setComments((comments) => {
+          return [optimisticComment, ...comments];
+        });
+        setCheck(false);
+        postComment(article_id, newComment).then();
```

Comment:
Great optimistic rendering, it would be good to revese this and give some feedback if the request fails.